flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -389,6 +394,36 @@ public String cancelWithSavepoint(JobID jobId, @Nullable String savepointDirecto
 		});
 	}
 
+	@Override
+	public Map<String, Object> getAccumulators(JobID jobID) throws Exception {
+		return getAccumulators(jobID, ClassLoader.getSystemClassLoader());
+	}
+
+	@Override
+	public Map<String, Object> getAccumulators(final JobID jobID, ClassLoader loader) throws Exception {
+		final JobAccumulatorsHeaders accumulatorsHeaders = JobAccumulatorsHeaders.getInstance();
+		final JobAccumulatorsMessageParameters accMsgParams = accumulatorsHeaders.getUnresolvedMessageParameters();
+		accMsgParams.jobPathParameter.resolve(jobID);
+		accMsgParams.includeSerializedAccumulatorsParameter.resolve(Collections.singletonList(true));
+
+		CompletableFuture<JobAccumulatorsInfo> responseFuture = sendRequest(
+			accumulatorsHeaders,
+			accMsgParams
+		);
+
+		return responseFuture.thenApply((JobAccumulatorsInfo accumulatorsInfo) -> {
+			if (accumulatorsInfo != null && accumulatorsInfo.getSerializedUserAccumulators() != null) {
+				try {
+					return AccumulatorHelper.deserializeAccumulators(accumulatorsInfo.getSerializedUserAccumulators(), loader);
+				} catch (Exception e) {
+					log.error("Deserialize accumulators with customized classloader error : {}", e);
```

Comment:
Let us properly fail with an exception here.

Reply:
Moreover, when logging an exception you don't have to specify a `{}` placeholder.